Library/Formula/xmake.rb

```diff
@@ -0,0 +1,26 @@
+class Xmake < Formula
+  desc "The Automatic Cross-platform Build Tool"
+  homepage "https://github.com/waruqi/xmake"
+  url "https://github.com/waruqi/xmake/archive/v1.0.4.tar.gz"
+  mirror "http://tboox.net/release/xmake/xmake-v1.0.4.tar.gz"
+  sha256 "9d55ca647f6f00bebdb6d9499c6607b71cfa8f42b6539bc0d17a472108f89e60"
+  head "https://github.com/waruqi/xmake.git"
+
+  def install
+    # install to the output directory first
+    system "./install", "output"
+
+    # install the xmake scripts
+    (share/"xmake").install Dir["xmake/*"]
+
+    # install the xmake binary
+    libexec.install "output/share/xmake/xmake"
+    bin.install Dir[libexec/"*"]
+    bin.env_script_all_files(libexec, :XMAKE_PROGRAM_DIR =>"#{share}/xmake")
```

Comment:
Is your `brew` up-to-date? Works locally here:

```
    # install the xmake scripts
    pkgshare.install Dir["xmake/*"]
```
There does seem to be an unrelated and non-fatal build error in there though:
```
link libxmaker.a
link demo.b
make[3]: *** [demo.b] Error 1
make[2]: *** [SUB_PROS_../demo_all] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [SUB_PROS_xmake_all] Error 2
make: *** [all] Error 2
make xmake
make xmake
```